src/Xrefcheck/Verify.hs

```diff
@@ -357,6 +362,24 @@ checkExternalResource VerifyConfig{..} link
           Nothing -> False
         Left _ -> False
 
+    processBrackets se = case (fromException se :: Maybe ParseException) of
+      Nothing -> throwError $ ExternalResourceInvalidUri Nothing
+      Just pe ->
+        let peBundle = (\(ParseException bundle) -> bundle) pe
+            err :| _ = bundleErrors peBundle
+        in case err of
+          TrivialError position _ _ ->
+            let characterInQuestion = link `T.index` position
```

Comment:
Instead of inspecting megaparsec's exception to see if it fails on an "illegal character", wouldn't it be easier to check if the link contains an illegal character upfront (before attempting to use `mkURI`)?

Reply:
Ignore this comment - Whether a character is legal or illegal depends on in which part of the URL it is located. A character may be legal in the host string and illegal in the path. Letting `mkUri` do the heavy lifting here is indeed the right choice.